### PR TITLE
T-3733: added <link> tag processing for canonical tags

### PR DESCRIPTION
**Pull request by mgrgic**
The html_content_parser is extended by checking <link> tags for canonical entries and avoids multiple canonical tags in html head.

### CODE REVIEW

File: composition/html_content_parser.go [hunk 1]
```diff
@@ -353,11 +357,43 @@ func processMetaTag(tagName string, attrs []html.Attribute, metaMap map[string]s
...
+	if (len(attrs) == 0) {
+		return true
+	}
+
+        const canonical = "canonical"
+        key := ""
+        value := ""
+
+        // e.g.: <link rel="canonical" href="/baumarkt/suche"> => key = canonical; val = /baumarkt/suche
+        for i := 0; i < len(attrs); i++ {
```
Comment (domano) [inline composition/html_content_parser.go]: How about "for attr := range attrs" instead?

**Comment by domano** [comment]
+1

**Comment by domano** [comment]
Whats the status on this branch?

**Comment by smancke** [comment]
Hey @mgrgic, Whats the status on this branch?

**Comment by smancke** [comment]
Do I understand it right, that the deduplication of the canonicals are only done while parsing one html?
What if multiple pages are merged and each brings a canonical with them?

**Comment by mgrgic** [comment]
The ParseHeadFragment() is just called in processMetaPriorityParsing(), where the priority is implemented.

**Comment by smancke** [comment]
ahh, ok, right.
Maybe it would make sense to move it to the place, where it's used.

**Pull request merged**